web/src/api/system/software.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) [2024-2025] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+type System = {
+  addons: AddonInfo[];
+  patterns: Pattern[];
+  repositories: Repository[];
+};
+
+type Pattern = {
+  /** Pattern name (internal ID) */
+  name: string;
+  /** Pattern category */
+  category: string;
+  /** User visible pattern name */
+  summary: string;
+  /** Long description of the pattern */
+  description: string;
+  /** {number} order - Display order (string!) */
+  order: number;
+  /** Icon name (not path or file name!) */
+  icon: string;
+  /** Whether the pattern if selected and by whom */
+  selectedBy?: SelectedBy;
+};
+
+type Repository = {
+  repo_id: number;
+  alias: string;
+  name: string;
+  raw_url: string;
+  product_dir: string;
+  enabled: boolean;
+  loaded: boolean;
+};
+
+type AddonInfo = {
+  id: string;
+  version: string;
+  label: string;
+  available: boolean;
+  free: boolean;
+  recommended: boolean;
+  description: string;
+  type: string;
+  release: string;
+};
+
+/**
+ * Enum for the reasons to select a pattern
+ */
+enum SelectedBy {
```

Comment:
this looks strange. Why it is duplicitely also at https://github.com/agama-project/agama/pull/2902/files#diff-baa3da095b381d915a0bcd93c0c2a72395b98c122a2b8beea3f2bf4436ec2004R30 ?

Reply:
As mentioned, these types will be revisited.